stbt/__init__.py

```diff
@@ -431,6 +431,58 @@ def is_screen_black(frame=None, mask=None, threshold=None):
     return _dut.is_screen_black(frame, mask, threshold)
 
 
+class FrameObject(_stbt.core.FrameObject):
+    """
+    A Frame Object is a class that extracts information from a frame of video,
+    typically by calling ``stbt.ocr()`` or ``stbt.match()``. All the rest of
+    your testing code then uses these objects. A Frame Object translates from
+    the vocabulary of low-level image processing functions and regions (like
+    ``stbt.ocr(region=stbt.Region(213, 23, 200, 36)))`` to the vocabulary of
+    high-level features and user-facing concepts (like ``programme_title``).
+
+    For more information about frame objects see
+    https://stb-tester.com/blog/2015/09/08/meet-the-frame-object-pattern
+
+    This base class is provided to make creating well-behaved frame objects
+    easier.  It defines:
+
+    *  __init__ - optionally taking a frame in the constructor
+    * __nonzero__ - based on the property is_visible which derived classes must
+      define.  This means the class will only be considered ``True`` if it's
+      visible, and so the other properties are valid.  This makes it easy to use
+      with ``wait_until``.
+    * __repr__ - including all the user-defined properties.  This makes using
+      the object in doctests convenient
+    * __hash__ and __cmp__ - If all the properties match between two instances
+      of a FrameObject then they are considered equal, even if the underlying
+      frame is different.  This can be useful for detecting changes or waiting
+      for a frame to stop changing before interrogating it.
+
+    This ticks three of the boxes in the `Frame Objects checklist`(
+    https://stb-tester.com/blog/2015/09/08/meet-the-frame-object-pattern)
+    allowing frame objects to be defined fairly minimally:
+
```

Comment:
Ideally this documentation would be more self-contained and wouldn't need to refer to the blogpost twice.

It's a shame it doesn't include the documentation in `test_frame_object.py` as this documentation here is what shows up in our python api docs at <https://stb-tester.com/manual/python-api>. Not sure what we could do about that, define the entire `Dialog` example frame object here in the docstring?